### PR TITLE
Remove System.Array type data

### LINKED ISSUE #3222: Remove type data for `System.Array` type
Type data for `System.Array` type contains single alias property `Count` -> `Length`. It was good for consistency, as `ICollection.Count` is implemented explicitly by arrays and was not visible while other collections types do not have `Length` property. But since PowerShell v3 it is possible to reference explicitly implemented interface members. So, this alias property no longer necessary. It also cause some headaches with `ConvertTo-Json` cmdlet:
```powershell
PS> ConvertTo-Json (,1)
[
  1
]
PS> ConvertTo-Json ([PSObject](,1))
{
  "value": [
    1
  ],
  "Count": 1
}
PS> Remove-TypeData System.Array
PS> ConvertTo-Json (,1)
[
  1
]
PS> ConvertTo-Json ([PSObject](,1))
[
  1
]
```

### PR DESCRIPTION
**Pull request by PetSerAl**
Closes #3222

### CODE REVIEW

**Review by daxian-dbw — APPROVED** [review]

**Comment by daxian-dbw** [comment]
#3230 is opened to track other `*.generated.cs` files.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/typedata.tests.ps1 [hunk 1]
```diff
@@ -1,19 +1,23 @@
 ﻿Describe "TestData cmdlets" -Tags "CI" {
     Context "Get-TypeData" {
         It "System.Array" {
-            (Get-TypeData System.Array).TypeName | Should Be System.Array
+            Get-TypeData System.Array | Should Be $null
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/typedata.tests.ps1]: Why do we need this test? Maybe remove?

Comment (PetSerAl) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/typedata.tests.ps1]: I just add it by analogy to [other](https://github.com/PowerShell/PowerShell/pull/3231/files/fd09cd7804fb760d08d9d39e172554c82f38f1a9#diff-7cd0edfab7610e4c4c8e59d2b167c885L24) existing test. If you think, that it is not necessary here, then I can remove that last commit.

Comment (daxian-dbw) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/typedata.tests.ps1]: I removed this test using on-page edit.

Comment (PetSerAl) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/typedata.tests.ps1]: @daxian-dbw You could just [remove](https://github.com/PowerShell/PowerShell/compare/80c03762597de92572403bb090ef7f01984a6cf7...0a1c96d4917ca09cd2e4a5ecd9b5b0c9591521c5) one last commit.

Comment (daxian-dbw) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/typedata.tests.ps1]: @PetSerAl Yes. I made this change with the online editor. But I'm going to squash and merge, so it should be fine.

**Comment by daxian-dbw** [comment]
@PetSerAl Thanks for the contribution!

**Pull request merged**